src/movies.js

```diff
@@ -1,16 +1,150 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(array) {
+  let directors = array.map((movie) => {
+    return movie.director;
+  });
+  return directors;
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array) {
+  let spielbergMovie = array.filter((movie) => {
+    return (
+      movie.director === "Steven Spielberg" &&
+      movie.genre.indexOf("Drama") != -1
+    );
+  });
+  return spielbergMovie.length;
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(array) {
+  if (array.length !== 0) {
+    let total = array.reduce((acc, movie) => {
+      if (movie.rate != "" && movie.rate != undefined) {
```

Comment:
you could reduce this conditionals to a single one (movie.rate) this will essentially check if movie.rate is not equal to ANY falsy value (0, "", undefined, null, NaN). and this way is commonly read as... "if movie.rate exists"
`if (movie.rate) {...}`